--- safetyhook/meson.build
+++ safetyhook/meson.build
@@ -0,0 +1,15 @@
+project('safetyhook', 'c', 'cpp', version: '0.1.3')
+
+safetyhook_inc = include_directories('.')
+safetyhook_lib = static_library(
+       'safetyhook',
+       include_directories: safetyhook_inc,
+       sources: [
+           'safetyhook.cpp',
+           'Zydis.c'
+       ]
+)
+safetyhook_dep = declare_dependency(
+       link_with: safetyhook_lib,
+       include_directories: safetyhook_inc,
+)
