+++ zlib/meson.build
@@ -0,0 +1,31 @@
+project('zlib', 'c', version: '1.0.0')
+
+zlib_inc = include_directories('.')
+zlib_sources = files(
+  'adler32.c',
+  'compress.c',
+  'crc32.c',
+  'deflate.c',
+  'gzclose.c',
+  'gzlib.c',
+  'gzread.c',
+  'gzwrite.c',
+  'infback.c',
+  'inffast.c',
+  'inflate.c',
+  'inftrees.c',
+  'trees.c',
+  'uncompr.c',
+  'zutil.c'
+)
+
+zlib_lib = static_library(
+  'zlib',
+  zlib_sources,
+  include_directories: zlib_inc
+)
+
+zlib_dep = declare_dependency(
+  include_directories: zlib_inc,
+  link_with: zlib_lib
+)
